src/v/config/configuration.cc

```diff
@@ -597,6 +597,13 @@ configuration::configuration()
       "How often do we trigger background compaction",
       {.needs_restart = needs_restart::no, .visibility = visibility::user},
       10s)
+  , log_disable_housekeeping_for_tests(
+      *this,
+      "log_disable_housekeeping_for_tests",
+      "Disables the housekeeping loop for local storage. The property exists "
+      "to simplify testing and shouldn't be set in production.",
+      {.needs_restart = needs_restart::yes, .visibility = visibility::tunable},
+      false)
```

Comment:
would it make sense to instead do something like set `compaction_interval_ms = infinitity`?

Reply:
Yea that'd probably work. in general i've erred on the side of explicit configs since it seems more obvious to use, and a bit more foolproof. If you feel strongly about it i can remove this

Reply:
sgtm